backend/src/main/java/co/inventorsoft/scripty/model/dto/FileNode.java

```diff
@@ -0,0 +1,31 @@
+package co.inventorsoft.scripty.model.dto;
+
+import java.io.Serializable;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.experimental.FieldDefaults;
+
+/**
+ * @author lzabidovsky 
+ */
+@AllArgsConstructor
+@Getter
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class FileNode implements Serializable {
+	private static final long serialVersionUID = 1L;
+
+	String parent;
+	String path;
+	String name;
+	String type;
```

Comment:
If you have a type field this becomes more generic than `FileNode`. E.g. I can create instance of this class this way:
`new FileNode("src", "src/main", "main", "directory")` which will represent directory. This is confusing and I don't see a reason why you cannot call it `Node`